src/controls/fabric_choice.rs

```diff
@@ -0,0 +1,65 @@
+use hashbrown::HashMap;
+use iced::{Alignment, Length};
+use iced_wgpu::Renderer;
+use iced_winit::Color;
+use iced_winit::widget::{Button, Row, Text};
+use crate::build::tenscript::fabric_plan;
+use crate::controls::{Action, Message};
+
+#[derive(Clone, Debug)]
+pub enum FabricChoiceMessage {
+    ChooseFabric(&'static str),
```

Comment:
Hmm, I wonder if you want to make these Strings instead to select runtime created fabric plans.

Reply:
Seems to add a lot of noise where it's used